### PR TITLE
Update container resources in ContainerStatusResponse

### PR DESCRIPTION
**Pull request by hasan4791**
#### What type of PR is this?
/kind other

#### What this PR does / why we need it:
Return resource status in ContainerStatusResponse

#### Which issue(s) this PR fixes:
Fixes https://github.com/cri-o/cri-o/issues/5653

#### Does this PR introduce a user-facing change?
Yes

```release-note
Added support to return ContainerResources in ContainerStatusResponse.
```

### CODE REVIEW

**Comment by hasan4791** [comment]
> 

Thank you for quick look, Let me check on that too.
Btw do we need linux host to run the prerequisite commands like(make lint or compile the binary & check it)

**Comment by haircommander** [comment]
lint I don't think so, compile I'm pretty sure to get real feedback (it should cross compile but lots of code is linux only)

**Comment by hasan4791** [comment]
@haircommander would you give me a small hint on the container object & state, Im in early stages on getting familiar with crio language 👍🏽

**Comment by haircommander** [comment]
it's defined in `internal/oci/container.go` and can be initialized when the container object is created. Then, a getter can be created and called

**Comment by hasan4791** [comment]
@haircommander container object already has the spec which has the resource status. Still do we need to maintain it on container object?

**Comment by haircommander** [comment]
> @haircommander container object already has the spec which has the resource status. Still do we need to maintain it on container object?

yeah the idea is to reduce the number of objects created and destroyed, and reduce the CPU of convering the objects. Trading the bit of memory to keep this information and reduce CPU has been the direction we've been going with these more frequenly called operations

**Comment by hasan4791** [comment]
@haircommander Could you please take a look at the changes.

File: server/container_status.go
Comment (haircommander) [inline server/container_status.go:61]: this could be done in one step

File: internal/oci/container.go
Comment (haircommander) [inline internal/oci/container.go:700]: just had this thought: what if we combined this with `SetSpec`? then we wouldn't have to remember to call each time SetSpec was called

Comment (hasan4791) [reply internal/oci/container.go:700]: makes sense

File: server/container_status.go
Comment (hasan4791) [reply server/container_status.go]: Done

File: internal/oci/container.go
Comment (hasan4791) [reply internal/oci/container.go:701]: Done

**Comment by haircommander** [comment]
would you mind squashing your commits?

**Comment by haircommander** [comment]
/approve

LGTM, @cri-o/cri-o-maintainers PTAL

**Comment by hasan4791** [comment]
@haircommander Aren;t we supposed to wait for this PR to be merged?
https://github.com/cri-o/cri-o/issues/5653#issuecomment-1162805557

File: internal/oci/container.go
Comment (jrvaldes) [inline internal/oci/container.go:79]: @hasan4791 Please add a footer in the commit message to reference the issue this PR is fixing. See [contribution guidelines](https://github.com/cri-o/cri-o/blob/main/CONTRIBUTING.md#submitting-pull-requests). 

For example:
```
Closes #5653
```

In addition, consider adding to the commit message what this commit is fixing and why.

File: internal/oci/container.go
Comment (jrvaldes) [inline internal/oci/container.go:750]: Consider adding a unit test for the newly introduced function, it will also make code coverage happy.

**Review by jrvaldes — CHANGES_REQUESTED** [review]
@hasan4791  Thanks for putting this together. PTAL at my comments.

**Comment by TomSweeneyRedHat** [comment]
Other than the remaining comments, LGTM

File: internal/oci/container.go
Comment (hasan4791) [reply internal/oci/container.go:750]: Done :)

File: internal/oci/container.go
Comment (hasan4791) [reply internal/oci/container.go:79]: Sure, will keep it in mind

**Comment by hasan4791** [comment]
@haircommander Please take a look once !!

**Comment by haircommander** [comment]
still LGTM, @cri-o/cri-o-maintainers PTAL

**Review by jrvaldes — APPROVED** [review]
/lgtm

**Comment by jrvaldes** [comment]
@hasan4791  Does this needs to be backported?

cc @haircommander

**Comment by haircommander** [comment]
/retest

> @hasan4791 Does this needs to be backported?

I don't believe so, as the values weren't used until in place VPA support was added which happened this release

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Pull request merged**
